internal/torcx/perform.go

```diff
@@ -327,6 +337,36 @@ func unpackTgz(applyCfg *ApplyConfig, tgzPath, imageName string) (string, error)
 	return topDir, nil
 }
 
+// mountSquashfs mounts a squashfs rootfs, returning the mounted directory.
+func mountSquashfs(applyCfg *ApplyConfig, archivePath, imageName string) (string, error) {
+	if applyCfg == nil {
+		return "", errors.New("missing apply configuration")
+	}
+
+	if archivePath == "" || imageName == "" {
+		return "", errors.New("missing unpack source")
+	}
+
+	topDir := filepath.Join(applyCfg.RunUnpackDir(), imageName)
+	if _, err := os.Stat(topDir); err != nil && os.IsNotExist(err) {
+		if err := os.MkdirAll(topDir, 0755); err != nil {
+			return "", err
+		}
+	}
+
+	loopDev, err := loopback.AttachLoopDevice(archivePath)
+	if err != nil {
+		return "", err
+	}
+	defer loopDev.Close()
+
+	if err := unix.Mount(loopDev.Name(), topDir, "squashfs", unix.MS_RDONLY, ""); err != nil {
```

Comment:
We should make sure that `/run` is mounted shared-recursive, in case anyone wants to bind-mount it in to a container.

Reply:
Nevermind - this runs so early on, it doesn't really matter.